### PR TITLE
Autonomous heading

### PR DESCRIPTION
**Pull request by AidanSmith**
This should make the heading sensor work to keep the robot straight while driving forward in autonomous. If you can thing of a better way to do this, please tell me.

Most of the PID code was just copied from my drive code, so it is no way calibrated. I added tuning for turning on and off whether to use PID during autonomous.

### CODE REVIEW

**Comment by AidanSmith** [comment]
I am currently testing on the emulator.

**Comment by AidanSmith** [comment]
Does NOT work in emulator, but close. I am working on it.

**Comment by AidanSmith** [comment]
It now works.

**Comment by AidanSmith** [comment]
That's all of the changes, I think.

**Comment by AidanSmith** [comment]
Ok, I fixed all of the bugs, one at a time, so now my new autonomous mode runs correctly on the emulator.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Autonomous.java [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+        FloatInput i = Mixing.select(calibrating, FloatMixing.division.of(
+                FloatMixing.multiplication.of(p, (FloatInput) iconstant), (FloatInput) period), FloatMixing.always(0));
+        FloatInput d = Mixing.select(calibrating, FloatMixing.multiplication.of(
+                FloatMixing.multiplication.of(p, (FloatInput) dconstant), (FloatInput) period), FloatMixing.always(0));
+
+        PIDControl pid = new PIDControl(HeadingSensor.absoluteYaw, desiredAngle, p, i, d);
+        pid.setOutputBounds(-1f, 1f);
+        pid.setIntegralBounds(-.5f, .5f);
+        Igneous.duringAuto.send(pid);
+        PIDValue = FloatMixing.createDispatch(Mixing.select(usePID, FloatMixing.always(0), pid), FloatMixing.onUpdate((FloatInput) pid));
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Autonomous.java]: I don't think you need `createDispatch` here.

Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Autonomous.java]: Also, `PIDValue` doesn't follow capitalization best practices.

**Comment by celskeggs** [comment]
Besides those, looks good.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Autonomous.java (hunk 1 above)
Comment (AidanSmith) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Autonomous.java]: On the emulator, when I don't have `createDispatch` the PID doesn't update until I change `usePID`.

**Comment by celskeggs** [comment]
Looks good.

**Pull request merged**